include/zephyr/drivers/gpio.h

```diff
@@ -1236,7 +1248,25 @@ static inline int z_impl_gpio_port_get_raw(const struct device *port,
 	const struct gpio_driver_api *api =
 		(const struct gpio_driver_api *)port->api;
 
-	return api->port_get_raw(port, value);
+	int ret = api->port_get_raw(port, value);
+
+	if (ret != 0) {
+		return ret;
+	}
+
+	int counter = 0;
+	gpio_port_value_t val_cp = *value;
+
+	while (val_cp != 0) {
+		if ((val_cp & 0x1) == 1) {
+			sys_port_trace_gpio_pin_active(port, (gpio_pin_t)counter);
```

Comment:
I wonder if the compiler optimizes out this loop if tracing is disabled?

Reply:
Probably could surround it with `if (IS_ENABLED(CONFIG_TRACING_GPIO)) { ... }`